de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,464 @@
+		editingDomain.getCommandStack().execute(importer.importSei(editingDomain, sei, mat));
+
+		//checks that if the same sei is imported and everything is empty, it stays empty
+		EList<APropertyInstance> caSei = sei.getCategoryAssignments().get(0).getPropertyInstances();
+		EList<APropertyInstance> caSei2 = sei2.getCategoryAssignments().get(0).getPropertyInstances();
+		assertTrue("same number of elements", caSei.size() == caSei2.size());
+		assertEquals("same testString", tc.getTestString(), tc1.getTestString());
+		assertEquals("same testBool", tc.getTestBool(), tc1.getTestBool());
+		assertEquals(tc.getTestFloat(), tc1.getTestFloat(), 0);
+		assertEquals("same testResource", tc.getTestResource(), tc1.getTestResource());
+		assertEquals("same testEnum", tc.getTestEnum(), tc1.getTestEnum());
+	}
+
+	@Test
+	public void testImportOfValuesChangeNothingValues() throws IOException {
+		TestCategoryAllProperty tc = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc);
+		tc.setTestBool(TEST_BOOL);
+		tc.setTestFloat(TEST_FLOAT);
+		tc.setTestEnum(TEST_ENUM_VALUE);
+		tc.setTestString(TEST_STRING);
+		tc.setTestInt(TEST_INT);
+		URI testUri = URI.createPlatformPluginURI("Testresource", true);
+		tc.setTestResource(testUri);
+		mat = exporter.exportSei(sei);
+
+		TestStructuralElement tsei2 = new TestStructuralElement(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName(TEST_SEI);
+		sei2.setUuid(sei.getUuid());
+		TestCategoryAllProperty tc1 = new TestCategoryAllProperty(testConcept);
+		tc1.setTestBool(TEST_BOOL);
+		tc1.setTestFloat(TEST_FLOAT);
+		tc1.setTestEnum(TEST_ENUM_VALUE);
+		tc1.setTestString(TEST_STRING);
+		tc1.setTestInt(TEST_INT);
+		tc1.setTestResource(testUri);
+		tsei2.add(tc1);
+		editingDomain.getCommandStack().execute(importer.importSei(editingDomain, sei, mat));
+
+		//checks if the same seiis imported and everything has a value, the value doesnt change
+		EList<APropertyInstance> caSei = sei.getCategoryAssignments().get(0).getPropertyInstances();
+		EList<APropertyInstance> caSei2 = sei2.getCategoryAssignments().get(0).getPropertyInstances();
+		assertTrue("same number of elements", caSei.size() == caSei2.size());
+		assertEquals("same testString", TEST_STRING, tc1.getTestString());
+		assertEquals("same testBool", TEST_BOOL, tc1.getTestBool());
+		assertEquals(TEST_FLOAT, tc1.getTestFloat(), 0);
+		assertEquals("same testResource", tc.getTestResource(), tc1.getTestResource());
+		assertEquals("same testEnumValue", tc.getTestEnumBean().getValue(), tc1.getTestEnumBean().getValue());
+		assertEquals("same testEnumValue", tc.getTestEnumBean().getUnit(), tc1.getTestEnumBean().getUnit());
+		assertEquals("same testInt", TEST_INT, tc1.getTestInt());
+	}
+
+	@Test
+	public void testImportOfValuesRef() throws IOException {
+		//Reference test target
+		final TestCategoryAllProperty TEST_REFERENCE_TARGET = new TestCategoryAllProperty(testConcept);
+		
+		//empty and import empty

```

Comment:
Missleading comment... Test is further down. Rather "Prepare test data" or remove it

Reply:
Done